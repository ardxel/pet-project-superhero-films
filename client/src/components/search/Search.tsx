import Loading from '@common/loading/Loading';
import { disableScroll, enableScroll } from '@common/tools/scroll-lock';
import IMovie from '@models/Movie';
import { fetchMoviesBySearch } from '@reduxproj/api/movie.api';
import { FC, useEffect, useRef, useState } from 'react';
import SearchCardMovie from './search-card-movie/SearchCardMovie';
import styles from './search.module.scss';

interface SearchProps {
  isOpen: boolean;
  setIsOpen: (arg: boolean) => void;
}

const Search: FC<SearchProps> = ({ isOpen, setIsOpen }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [movies, setMovies] = useState<IMovie[] | null>(null);
  // const { data, isLoading, error } = useFetch(fetchMoviesBySearch, {
  //   search: searchTerm,
  // });

  useEffect(() => {
    fetchMoviesBySearch({ search: searchTerm })
      .then((data) => setMovies(data?.movies as IMovie[]))
      .catch((error) => console.log(error));
  }, [searchTerm]);

  const inputRef = useRef<HTMLInputElement>(null);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(e.target.value);
  };
  useEffect(() => {
    if (isOpen && searchTerm) {
      disableScroll();
    }
    if (isOpen && !searchTerm) {
      inputRef.current!.focus();
      enableScroll();
    }
    if (!isOpen) {
      enableScroll();
    }
  }, [isOpen, searchTerm]);
  return (
    <>
      <div
        className={styles.searchModal}
        data-destiny="search"
        style={{ display: isOpen ? 'block' : 'none' }}>
        <section className={styles.field}>
          <input
            ref={inputRef}
            type="text"
            placeholder="search..."
            onChange={handleChange}
          />
        </section>
      </div>
      {/* background blur element */}
      {searchTerm && isOpen && <div className={styles.bg}></div>}

      {searchTerm && isOpen && (
        <div className={styles.results}>
          <div className={styles.container}>
            {isLoading && <Loading />}
            {movies?.map((movie) => {
              return (
                <SearchCardMovie
                  key={movie._movieId}
                  {...movie}
                  setIsOpen={setIsOpen}
                />
              );
            })}
          </div>
        </div>
      )}
    </>
  );
};

export default Search;
