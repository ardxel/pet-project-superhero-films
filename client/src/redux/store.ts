import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { newsReducer } from '@reduxproj/reducers/news.reducer';
import { userReducer } from '@reduxproj/reducers/user.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  news: newsReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware(),
});

export default store;
export type AppStore = typeof store;
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
