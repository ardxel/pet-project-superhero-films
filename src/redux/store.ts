import { configureStore } from '@reduxjs/toolkit';
import {movieSlice} from 'redux/reducers/movieReducer';
import { moviesApi } from 'redux/api/moviesApi';

const store = configureStore({
  reducer: {
    [movieSlice.name]: movieSlice.reducer,
    [moviesApi.reducerPath]: moviesApi.reducer,
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(moviesApi.middleware)
  }
});

export default store;
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;


