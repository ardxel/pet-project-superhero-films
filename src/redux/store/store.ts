import { configureStore } from '@reduxjs/toolkit';
import {movieSlice} from '../reducers/movieReducer';

const store = configureStore({
  reducer: {
    [movieSlice.name]: movieSlice.reducer
  }
});

export default store;
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;
