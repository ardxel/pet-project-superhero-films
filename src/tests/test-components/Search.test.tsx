import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Search from '@components/search/Search';
import store from '@reduxproj/store';
import { Provider } from 'react-redux';
describe('<Search> component test', () => {
  it('should render input for searching', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Search isOpen setIsOpen={() => true} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
