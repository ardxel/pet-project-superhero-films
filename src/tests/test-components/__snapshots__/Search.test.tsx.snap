// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> component test should render input for searching 1`] = `
<div
  className="searchModal"
  data-destiny="search"
  style={
    {
      "display": "block",
    }
  }
>
  <section
    className="field"
  >
    <input
      onChange={[Function]}
      placeholder="search..."
      type="text"
    />
  </section>
</div>
`;
